raft/src/main/java/org/apache/kafka/snapshot/FileRawSnapshotWriter.java

```diff
@@ -53,13 +56,20 @@ public long sizeInBytes() throws IOException {
     }
 
     @Override
-    public void append(ByteBuffer buffer) throws IOException {
+    public void append(BaseRecords records) throws IOException {
         if (frozen) {
             throw new IllegalStateException(
-                String.format("Append is not supported. Snapshot is already frozen: id = %s; temp path = %s", snapshotId, tempSnapshotPath)
+                    String.format("Append is not supported. Snapshot is already frozen: id = %s; temp path = %s", snapshotId, tempSnapshotPath)
             );
         }
-
+        ByteBuffer buffer;
+        if (records instanceof MemoryRecords) {
+            buffer = ((MemoryRecords) records).buffer();
+        } else {
+            buffer = ByteBuffer.allocate(records.sizeInBytes());
+            ((FileRecords) records).channel().read(buffer);
+            buffer.flip();
+        }
```

Comment:
What do you think about keeping this signature as `void append(ByterBuffer buffer)`, having a static function that knows how to convert a `BaseRecords` to `ByteBuffer` and moving this responsibility to the `KafkaRaftClient`?

If I am not mistake, as of right now it is only the `KafkaRaftClient` that needs to deal with this mismatch.

Reply:
> I change the signature to keep consistent with FileRawSnapshotReader

Okay. I think this is something that I struggled with when creating the original APIs. I am okay with "inconsistent" APIs since `RawSnapshot{Reader,Writer}` are internal interfaces to the raft client and are not exposed to the state machine (controller).

I think this "inconsistency" will go away when we implement the long term solution.

Reply:
Done, make parameter type to MemoryRecords.

Reply:
Keep 2 methods here, one for MemoryRecords and one for UnalignedMemoryRecords.